docs/operate/get-started/other-hardware/_index.md

```diff
@@ -51,48 +51,68 @@ For C++ module examples, see the [C++ examples directory on GitHub](https://gith
 {{% /alert %}}
 
 {{< expand "How and where do modules run?" >}}
-Modules run on your machine, alongside [`viam-server`](/operate/reference/viam-server/) as separate processes, communicating with `viam-server` over UNIX sockets.
-
-When a module initializes, it registers its {{< glossary_tooltip term_id="model" text="model or models" >}} and associated [APIs](/dev/reference/apis/) with `viam-server`, making the new model available for use.
-`viam-server` manages the [dependencies](/operate/reference/viam-server/#dependency-management), [start-up](/operate/reference/viam-server/#start-up), [reconfiguration](/operate/reference/viam-server/#reconfiguration), [data management](/data-ai/capture-data/capture-sync/), and [shutdown](/operate/reference/viam-server/#shutdown) behavior of your modular resource.
+Modules run on your machine, alongside `viam-server` as separate processes, communicating with `viam-server` over UNIX sockets.
+
+[`viam-server` manages](/operate/reference/viam-server/) the dependencies, start-up, reconfiguration, [data management](/data-ai/capture-data/capture-sync/), and shutdown behavior of your modular resource.
+
+The lifecycle of a module and the resources it provides is as follows:
+
+1. `viam-server` starts, and if it is connected to the internet, it checks for configuration updates.
+1. `viam-server` starts any configured modules.
+1. When a module initializes, it registers its model or models and associated [APIs](/dev/reference/apis/) with `viam-server`, making the models available for use.
+1. For each modular resource configured on the machine, `viam-server` uses the resource's `validate` function and the `depends_on` field in the resource configuration to determine the dependencies of the resource.
+1. If a dependency is not already running, `viam-server` starts it before starting the resource.
+   If a dependency is not found or fails to start, `viam-server` will not start the resource that depends on it.
+1. `viam-server` builds the resource based on its resource configuration.
+1. If configuration fails due to a validation failure or an exception thrown by the reconfigure function, `viam-server` attempts to reconfigure the resource.
```

Comment:
```suggestion
1. `viam-server` calls the resource's constructor to build the resource based on its configuration.
   Typically, the constructor calls the reconfigure function.
1. If configuration fails due to a validation failure or an exception thrown by the reconfigure or constructor functions, `viam-server` attempts to reconfigure the resource.
```
I realized we got the following question which I want to make sure this lifecycle explanation answers: 
"if constructor returns an error, will the resource manager try again repeatedly to register the camera by re-calling constructor?"
What's a good way to answer this @michaellee1019 ?

Reply:
the resource manager will try repeatedly if the constructor errors, yes